src/lib.rs

```diff
@@ -14,19 +14,193 @@ use rustty::CellAccessor;
+        // are in little endian order, but the bits within each byte are big
+        // endian:
+        // [00001010][00000111]
+        let bits = BitVec::from_bytes(&bytes);
+        let expected = vec![
+            // 0th byte representing 10
+            false,
+            false,
+            false,
+            false,
+            true,
+            false,
+            true,
+            false,
+            // 1st byte representing 7
+            false,
+            false,
+            false,
+            false,
+            false,
+            true,
+            true,
+            true];
+        for i in 0..16 {
+            assert_eq!(bits[i], expected[i]);
+        }
+    }
+
+    #[test]
+    fn test_conway_kode() {
+        let expected = "476348294852520375132009738840824718882889556\
+                        423255282629108876378472743729817205343700177\
+                        683429960362194923168607044012736510546282236\
+                        08960".parse::<BigInt>().unwrap();
+        assert_eq!(expected, conway_kode());
+    }
+
+    #[test]
+    fn test_grid_from_gidney_string() {
+
+        let grid = grid_from_gidney_string(vec!["   ", "   "]);
+        let mut expected = CellSet::new();
+        assert_eq!(grid, expected);
+
+        let grid = grid_from_gidney_string(vec!["#  ", "   "]);
+        expected.insert((0, 0));
+        assert_eq!(grid, expected);
+
+        let grid = grid_from_gidney_string(vec!["#  ", " # "]);
+        expected.insert((1, 1));
+        assert_eq!(grid, expected);
+    }
+
+    #[test]
+    fn test_decimal_encoded_string_to_bitvec() {
+        let s = "1802";
+        let bitvec = decimal_encoded_string_to_bitvec(s);
+        for i in 0..16 {
+            assert_eq!(bitvec[i], EXPECTED_1082_BITS[i]);
+        }
+    }
+
+    #[test]
+    fn test_bigint_to_bitvec(){
+        let v = bigint_to_bitvec(BigInt::from(1802));
+        for i in 0..16 {
+            assert_eq!(v[i], EXPECTED_1082_BITS[i]);
+        }
+    }
+
+    #[test]
+    fn test_get_state() {
+        let mut grid = CellSet::new();
+        //  0
+        // 0#< look here
+        //  ^
+        grid.insert((0, 0));
+        assert_eq!(get_state(&grid, &(0, 0)), 16); // 2^4
+        //  01
+        // 0#-< look here
+        // 1-#
+        //  ^
+        grid.insert((1, 1));
+        assert_eq!(get_state(&grid, &(0, 0)), 272); // 2^4 + 2^8
+    }
+}
+
+pub fn conway_kode() -> BigInt {
+    let mut kode = BigInt::from(0);
+    for state in 0..512 {
+        let mut bit_count = 0usize;
+        let current_state = (state >> 4) % 2;
+        for bit_offset in [0, 1, 2, 3, 5, 6, 7, 8].iter() {
+            bit_count += (state >> bit_offset) & 1usize;
+        }
+        let result = BigInt::from(match bit_count {
+            2 => current_state,
+            3 => 1,
+            _ => 0
+        });
+    kode = kode + (result << state);
+    }
+    kode
+}
+
+pub fn grid_from_gidney_string(s: Vec<&str>) -> CellSet {
```

Comment:
Like the above, this should be moved into the test module. We can move it out if we need it in the main code.